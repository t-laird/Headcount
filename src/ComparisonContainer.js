import React from 'react';
import './Styles/ComparisonContainer.css';
import Card from './Card';

function ComparisonContainer (props) {
  return (
    <div className="ComparisonContainer">
      <h3>Click Two Cards to Compare the School Districts</h3>
      <Card data={{}}/>
      <Card data={{}}/>
      <Card data={{}}/>
    </div>
  );
}

export default ComparisonContainer;