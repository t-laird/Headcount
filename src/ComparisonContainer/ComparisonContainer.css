.ComparisonContainer {
  display: flex;
  flex-direction: column; }
  .ComparisonContainer .instructions {
    display: flex;
    justify-content: space-around;
    margin-bottom: -12px;
    width: 690px; }
    .ComparisonContainer .instructions h5 {
      color: grey;
      font-style: italic;
      margin: 0px; }
  .ComparisonContainer .cards {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    margin: 15px auto 0;
    width: 780px;
    border: 1px dashed black; }
    .ComparisonContainer .cards h3 {
      color: #ff4d4d;
      width: 100%;
      text-align: center;
      font-size: 18pt;
      margin: 20px auto; }

@media screen and (max-width: 945px) {
  .ComparisonContainer {
    width: 700px; } }

@media screen and (max-width: 860px) {
  .ComparisonContainer {
    width: 520px; } }

@media screen and (max-width: 670px) {
  .ComparisonContainer {
    width: 460px; } }
