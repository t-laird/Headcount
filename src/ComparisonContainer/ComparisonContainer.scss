.ComparisonContainer {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 15px auto 0;
  width: 780px;
  border: 1px dashed black;
  h3 {
    color: lighten(red, 15%);
    width: 100%;
    text-align: center;
    font-size: 18pt;
  }
}

@media screen and (max-width: 945px){
  .ComparisonContainer {
    width: 700px;
  }
}

@media screen and (max-width: 860px){
  .ComparisonContainer {
    width: 520px;
  }
}

@media screen and (max-width: 670px){
  .ComparisonContainer {
    width: 460px;
  }
}