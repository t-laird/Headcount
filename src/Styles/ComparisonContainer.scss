.ComparisonContainer {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 0px auto;
  width: 780px;
  h3 {
    color: lighten(red, 15%);
    width: 100%;
  }
}

@media screen and (max-width: 830px){
  .ComparisonContainer {
    width: 520px;
  }
}

@media screen and (max-width: 570px){
  .ComparisonContainer {
    width: 260px;
  }
}