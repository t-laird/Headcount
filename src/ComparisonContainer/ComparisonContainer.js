import React from 'react';
import './ComparisonContainer.css';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

const ComparisonContainer = (props) => {
  const selectedCards = [];

  selectedCards[0] = populateComparison(0);
  selectedCards[1] = populateComparison(1);

  function populateComparison(position) {
    let comparisonIndex = 
      props.comparison[position] 
        ? props.cards.find(card => {
          const location = Object.keys(card)[0];

          return (props.comparison[position] === location);
        }) 
        : null;

    return comparisonIndex;
  }

  const mappedCards = selectedCards.map((card, index) => {
    if (!card) {
      return null;
    }
    const location = Object.keys(card)[0];

    return (
      <Card 
        key={`card-${index}`}
        selectCard={props.selectCard}
        data={card[location]}
        district={location}
        type='Card selected'
      />
    );
  });

  const comparisonCard = 
    (mappedCards[0] && mappedCards[1]) 
      ? props.compareCards(mappedCards[0].props.district,                               mappedCards[1].props.district) 
      : null;
  
  return (
    <div className="ComparisonContainer">
      <h3>Click Two Cards to Compare the School Districts</h3>
      {
        mappedCards[0] &&
        mappedCards[0]
      }
      {
        !mappedCards[0] &&
        <Card 
          type="Card comparison" 
          data={{}}
        />
      }
      
      {
        comparisonCard &&
        <Card 
          data={{}}
          type="Card displayComparedData"
          compareData={comparisonCard}
          district1={mappedCards[0].props.district.toUpperCase()}
          district2={mappedCards[1].props.district.toUpperCase()}
        />
      }
      {
        !comparisonCard &&
        <Card
          type="Card compared"
          data={{}}
        />
      }

      
      {
        mappedCards[1] &&
        mappedCards[1]
      }
      {
        !mappedCards[1] &&
        <Card type="Card comparison" data={{}}/>
      }
    </div>
  );
};

ComparisonContainer.propTypes = {
  cards: PropTypes.array,
  selectCard: PropTypes.func,
  comparison: PropTypes.array,
  compareCards: PropTypes.func
};

export default ComparisonContainer;