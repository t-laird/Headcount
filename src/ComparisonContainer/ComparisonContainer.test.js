import React from 'react';
import ReactDOM from 'react-dom';
import ComparisonContainer from './ComparisonContainer';
import { shallow, mount } from 'enzyme';

describe('ComparisonContainer test', () => {
  it('should render the correct components', () => {
    const renderedComparisonContainer = shallow(<ComparisonContainer cards={[]} comparison={[null, null]} />);
    expect(renderedComparisonContainer.find('.ComparisonContainer').length).toEqual(1);
  });

  it('should match the ComparisonContainer snapshot', () => {
    const renderedComparisonContainer = shallow(<ComparisonContainer cards={[]} comparison={[null, null]}/>);
    expect(renderedComparisonContainer).toMatchSnapshot();
  });

  it('It should render the correct default cards when not given comparison data', () => {
    const renderedComparisonContainer = mount(<ComparisonContainer cards={[]} comparison={[null, null]} />);
    expect(renderedComparisonContainer.find('Card').length).toEqual(3);
    expect(renderedComparisonContainer.find('.comparison').length).toEqual(2);
    expect(renderedComparisonContainer.find('.compared').length).toEqual(1);
  });

  it('Should render a standard card with a selected class but no middle comparison when passed a single valid comparison', () => {
    const renderedComparisonContainer = mount(<ComparisonContainer cards={[{Colorado: {2003: 0.5, 2004: 0.6}}]} comparison={['Colorado', null]} />);
    expect(renderedComparisonContainer.find('.selected').length).toEqual(1);
    expect(renderedComparisonContainer.find('.comparison').length).toEqual(1);
    expect(renderedComparisonContainer.find('.compared').length).toEqual(1);
  });

  it('Should render two standard cards with the selected class and a middle comparison card when passed two valid comparisons', () => {
    const mockFunc = jest.fn();
    mockFunc.mockReturnValue(true);
    const renderedComparisonContainer = 
      mount(<ComparisonContainer 
        cards={[{Colorado: {2003: 0.5, 2004: 0.6}}, {['ADAMS COUNTY 14']: {2003: 0.5, 2004: 0.6}}]} 
        comparison={['Colorado', 'ADAMS COUNTY 14']} 
        compareCards={mockFunc} 
      />);

    expect(mockFunc.mock.calls.length).toEqual(1);
    expect(renderedComparisonContainer.find('.selected').length).toEqual(2);
    expect(renderedComparisonContainer.find('.comparison').length).toEqual(0);
    expect(renderedComparisonContainer.find('.displayComparedData').length).toEqual(1);
    
  });
});