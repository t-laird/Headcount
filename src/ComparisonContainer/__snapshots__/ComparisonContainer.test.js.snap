exports[`ComparisonContainer test should match the ComparisonContainer snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    cards={Array []}
    comparison={
        Array [
            null,
            null,
          ]
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Click Two Cards to Compare the School Districts
</h3>,
        null,
        <Card
          data={Object {}}
          type="Card comparison" />,
        null,
        <Card
          data={Object {}}
          type="Card compared" />,
        null,
        <Card
          data={Object {}}
          type="Card comparison" />,
      ],
      "className": "ComparisonContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click Two Cards to Compare the School Districts",
        },
        "ref": null,
        "rendered": "Click Two Cards to Compare the School Districts",
        "type": "h3",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {},
          "type": "Card comparison",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {},
          "type": "Card compared",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {},
          "type": "Card comparison",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Click Two Cards to Compare the School Districts
</h3>,
          null,
          <Card
            data={Object {}}
            type="Card comparison" />,
          null,
          <Card
            data={Object {}}
            type="Card compared" />,
          null,
          <Card
            data={Object {}}
            type="Card comparison" />,
        ],
        "className": "ComparisonContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click Two Cards to Compare the School Districts",
          },
          "ref": null,
          "rendered": "Click Two Cards to Compare the School Districts",
          "type": "h3",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Object {},
            "type": "Card comparison",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Object {},
            "type": "Card compared",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Object {},
            "type": "Card comparison",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
